query-service-impl/src/main/java/org/hypertrace/core/query/service/postgres/converters/DefaultColumnRequestConverter.java

```diff
@@ -207,6 +211,63 @@ private boolean handleConversionForNullOrEmptyBytesLiteral(
     return false;
   }
 
+  private boolean isFilterForArrayColumnType(Filter filter, ColumnRequestContext context) {
+    return isSimpleAttributeExpression(filter.getLhs())
+        && filter.getRhs().getValueCase().equals(LITERAL)
+        && context.isArrayColumnType();
+  }
+
+  /** Handles value conversion of a array expression based */
+  private void handleConversionForArrayColumnExpression(
+      String lhs, String operator, Expression rhs, StringBuilder builder, Builder paramsBuilder) {
+    Value value = rhs.getLiteral().getValue();
+
+    if (handleConversionForNullOrEmptyArrayLiteral(lhs, operator, builder, value)) return;
+
+    builder.append(lhs);
+    // support only equals and IN operator
+    // both of them are handled as contains check
+    if (!operator.equals("=") && !operator.equals("IN")) {
```

Comment:
We may have to handle other operators too. But, as basic can't we support `not in` and `!=`?

Reply:
Done